From ee53a04aa75bf32f70f8c4d58168ca1c88e501c2 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sun, 2 Aug 2026 11:48:28 +0200
Subject: [PATCH] ARM: dts: allwinner: sun8i-a83t-tbs-a711: Describe camera
 mounting

Both sensors are mounted a quarter turn from the panel, in opposite
directions, and neither said so. Userspace had to guess.

rotation is the counter-clockwise angle between the sensor's readout
frame and the scene, so the picture comes out upright after turning it
back the other way: 90 for the rear hm5065, 270 for the front gc2145.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts b/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
index 6edeaf8ed431..099377539ac6 100644
--- a/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
+++ b/arch/arm/boot/dts/allwinner/sun8i-a83t-tbs-a711.dts
@@ -358,6 +358,9 @@ hm5065: rear-camera@1f {
 		reset-gpios = <&pio 4 18 GPIO_ACTIVE_LOW>; /* PE18 */
 		enable-gpios = <&pio 4 19 GPIO_ACTIVE_HIGH>; /* PE19 */
 
+		orientation = <1>;
+		rotation = <90>;
+
 		port {
 			hm5065_ep: endpoint {
 				remote-endpoint = <&csi_hm5065_ep>;
@@ -381,6 +384,9 @@ gc2145: front-camera@3c {
 		reset-gpios = <&pio 4 16 (GPIO_ACTIVE_LOW | GPIO_OPEN_DRAIN)>; /* PE16 */
 		enable-gpios = <&pio 4 17 (GPIO_ACTIVE_LOW | GPIO_OPEN_DRAIN)>; /* PE17 */
 
+		orientation = <0>;
+		rotation = <270>;
+
 		port {
 			gc2145_ep: endpoint {
 				remote-endpoint = <&csi_gc2145_ep>;
-- 
2.43.0

